src/clients/AcrossConfigStoreClient/AcrossConfigStoreClient.ts

```diff
@@ -417,6 +421,33 @@ export class AcrossConfigStoreClient extends BaseAbstractClient {
         if (!isNaN(args.value)) {
           this.cumulativeMaxRefundCountUpdates.push(args);
         }
+      } else if (args.key === utf8ToHex(GLOBAL_CONFIG_STORE_KEYS.CHAIN_ID_INDICES)) {
+        // First remove out all the spaces
+        const rawChainIndices = String(args.value).replace(/\s/g, "");
+        // Sanity check to verify that this is a string representation of an array of
+        // positive integers. Let's confirm this via a regex.
+        if (!rawChainIndices.match(/^\[[0-9,]+\]$/)) {
+          this.logger.warn({ at: "ConfigStore", message: `The array ${rawChainIndices} is invalid.` });
+          // If not a valid array, skip.
+          continue;
+        }
+        // Let's parse this via JSON.parse. Since we've passed the regex check, we can
+        // be sure that this is a valid array of positive integers.
+        const chainIndices = JSON.parse(rawChainIndices) as number[];
+        // We need to now check that all the indices exist and do not contain duplicates.
+        // This is essentially a set of indices that exist in the array length. For example,
+        // for a length of 5, the valid indices must contain [0, 1, 2, 3, 4].
+        // Let's check for this condition and for duplicates. If we find this to be invalid,
+        // we'll skip this update.
+        if (
+          chainIndices.length !== new Set(chainIndices).size ||
+          chainIndices.some((index) => index < 0 || index >= chainIndices.length)
+        ) {
+          this.logger.warn({ at: "ConfigStore", message: `The array ${chainIndices} is invalid.` });
+          continue;
+        }
+        // If all else passes, we can add this update.
+        this.chainIdIndicesUpdates.push({ ...args, value: chainIndices });
```

Comment:
can we add one more check: the UMIP states that chain ID's can only append chains. Can you add a check that the chain ID list only adds ID's to the previous list?

The default case is [1,10,137,288,42161] which we can store as a variable and we should use this as the "previous" list for the first update

Reply:
Done: https://github.com/across-protocol/sdk-v2/pull/346/commits/c3af2dff62d51862a10462a45d3696b6ca229b03